### ISSUE TITLE
Unknown variable name allowed through using it to calculate aggregate column

### ISSUE THREAD

**Issue by haydenflinner** [PRQL/prql#941]
```
from numbers
group [type] (
  aggregate [
   total_amt = sum amount,
   ]
  aggregate [
   max amount
  ]
)
```
compiles to

```
WITH table_0 AS (
  SELECT
    type,
    SUM(amount) AS total_amt
  FROM
    numbers
  GROUP BY
    type
)
SELECT
  type,
  MAX(amount)
FROM
  table_0
GROUP BY
  type
```

Which as I understand it should never work because amount is not a column in table_0. If I comment out the `SUM(amount) as total_amt` line, the compiler points out that `amount` is an unknown variable.

**Comment by aljazerzen** [comment]
Yes, this is a bug with current resolver implementation. It should throw an error, because the second `aggregate` references `amount` which is no longer present in the table (first `aggregate` converts the table to have only columns `type` and `total_amt`).

This issue will be fixed by #593, ETA 3 weeks.

**Comment by aljazerzen** [comment]
This does throw an error now. Took a bit longer than 3 weeks :D
